File: backend/app/schemas/review_translation.py
"""체험단 번역 모듈 Pydantic 스키마 (업무개선요구사항 #17)."""
from __future__ import annotations

import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, Field


class ReviewTranslationCreate(BaseModel):
    """POST /api/review-translations/translate 요청 본문."""

    source_text: str = Field(min_length=1, max_length=20_000, description="중국어 원문")
    campaign: str | None = Field(default=None, max_length=200, description="캠페인명")
    reviewer_name: str | None = Field(default=None, max_length=100, description="체험단명")
    platform: str | None = Field(default=None, max_length=50, description="플랫폼(샤오홍슈/웨이보 등)")


class ReviewTranslationUpdate(BaseModel):
    """PUT /api/review-translations/{id} 요청 본문. 모든 필드 옵션."""

    translated_text: str | None = Field(default=None, min_length=1, max_length=20_000)
    campaign: str | None = Field(default=None, max_length=200)
    reviewer_name: str | None = Field(default=None, max_length=100)
    platform: str | None = Field(default=None, max_length=50)


class ReviewTranslationRead(BaseModel):
    """단건 응답 스키마. 모델 ORM 객체를 직접 직렬화."""

    id: uuid.UUID
    created_at: datetime
    updated_at: datetime | None
    source_text: str
    translated_text: str
    campaign: str | None
    reviewer_name: str | None
    platform: str | None
    model_used: str
    input_tokens: int | None
    output_tokens: int | None
    cost_usd: Decimal | None
    created_by_id: uuid.UUID | None

    model_config = {"from_attributes": True}


class ReviewTranslationList(BaseModel):
    """페이지네이션 응답."""

    items: list[ReviewTranslationRead]
    total: int
    page: int
    page_size: int

File: frontend/src/api/reviewTranslation.ts
import api from "./client";

/**
 * 체험단 후기 중→한 번역 API 클라이언트 (업무개선요구사항 #17).
 *
 * 백엔드: `app/routers/review_translation.py`
 * 권한: `review_translation` 모듈 — admin + marketing_1.
 */

export interface ReviewTranslation {
  id: string;
  created_at: string;
  updated_at: string | null;
  source_text: string;
  translated_text: string;
  campaign: string | null;
  reviewer_name: string | null;
  platform: string | null;
  model_used: string;
  input_tokens: number | null;
  output_tokens: number | null;
  cost_usd: number | null;
  created_by_id: string | null;
}

export interface ReviewTranslationCreate {
  source_text: string;
  campaign?: string | null;
  reviewer_name?: string | null;
  platform?: string | null;
}

export interface ReviewTranslationUpdate {
  translated_text?: string;
  campaign?: string | null;
  reviewer_name?: string | null;
  platform?: string | null;
}

export interface ReviewTranslationListResponse {
  items: ReviewTranslation[];
  total: number;
  page: number;
  page_size: number;
}

export interface ReviewTranslationListParams {
  page?: number;
  page_size?: number;
  search?: string;
  campaign?: string;
}

export async function translateAndSave(
  body: ReviewTranslationCreate,
): Promise<ReviewTranslation> {
  const res = await api.post<ReviewTranslation>(
    "/api/review-translations/translate",
    body,
  );
  return res.data;
}

export async function listReviewTranslations(
  params?: ReviewTranslationListParams,
): Promise<ReviewTranslationListResponse> {
  const res = await api.get<ReviewTranslationListResponse>(
    "/api/review-translations",
    { params },
  );
  return res.data;
}

export async function getReviewTranslation(
  id: string,
): Promise<ReviewTranslation> {
  const res = await api.get<ReviewTranslation>(
    `/api/review-translations/${id}`,
  );
  return res.data;
}

export async function updateReviewTranslation(
  id: string,
  body: ReviewTranslationUpdate,
): Promise<ReviewTranslation> {
  const res = await api.put<ReviewTranslation>(
    `/api/review-translations/${id}`,
    body,
  );
  return res.data;
}

export async function deleteReviewTranslation(id: string): Promise<void> {
  await api.delete(`/api/review-translations/${id}`);
}
